src/Form/MediaLibraryEmbeddableForm.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace Drupal\stanford_media\Form;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\media\OEmbed\ResourceFetcherInterface;
+use Drupal\media\OEmbed\UrlResolverInterface;
+use Drupal\media_library\MediaLibraryUiBuilder;
+use Drupal\media_library\OpenerResolverInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\media_library\Form\OEmbedForm;
+
+/**
+ * Media library add stanford embed input form.
+ *
+ * @package Drupal\stanford_media\Form
+ */
+class MediaLibraryEmbeddableForm extends OEmbedForm {
+
+  /**
+   * The current User.
+   *
+   * @var Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The name of the oEmbed field.
+   *
+   * @var string
+   */
+  public $oEmbedField;
+
+  /**
+   * The name of the Unstructured field.
+   *
+   * @var string
+   */
+  public $unstructuredField;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, MediaLibraryUiBuilder $library_ui_builder, UrlResolverInterface $url_resolver, ResourceFetcherInterface $resource_fetcher, ConfigFactoryInterface $config_factory, AccountInterface $account, OpenerResolverInterface $opener_resolver = NULL) {
+    parent::__construct($entity_type_manager, $library_ui_builder, $url_resolver, $resource_fetcher, $opener_resolver);
+    $this->currentUser = $account;
+    $this->oEmbedField = $config_factory->get('media.type.embeddable')->get('source_configuration.oembed_field_name');
+    $this->unstructuredField = $config_factory->get('media.type.embeddable')->get('source_configuration.unstructured_field_name');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('media_library.ui_builder'),
+      $container->get('media.oembed.url_resolver'),
+      $container->get('media.oembed.resource_fetcher'),
+      $container->get('config.factory'),
+      $container->get('current_user'),
+      $container->get('media_library.opener_resolver')
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getFormId() {
+    return $this->getBaseFormId() . '_embeddable';
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function buildInputElement(array $form, FormStateInterface $form_state) {
+    // This was adapted from \Drupal\media_library\Form\OembedForm.
+    $authorized_for_unstructured = $this->currentUser->hasPermission('create field_media_embeddable_code') || $this->currentUser->hasPermission('edit field_media_embeddable_code');
```

Comment:
I'm thinking we do a form alter at the profile level to disable access based on field permissions.

Reply:
We control the permissions with configs in the profile, which makes it easy to turn it off and on.  The only benefit I see would be to remove the dependency on the `field_permissions` module, which is not a bad thing, but also not strictly necessary.

A downside on that would be if we want to change the setting, we then have to change it in `form_alter`s for both profiles, rather than just with a config setting, right?

Reply:
or just one form alter in the profile helper module. I think thats a more appropriate solution than having it in this module and forcing the dependency on the field permission module.